src/AwsServiceProvider.php

```diff
@@ -37,30 +44,39 @@ class AwsServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        $this->app['aws'] = $this->app->share(function ($app) {
-            // Retrieve the config
-            $config = $app['config']['aws'] ?: $app['config']['aws::config'];
-            if (isset($config['config_file'])) {
-                $config = $config['config_file'];
-            }
+        $this->app['aws'] = $this->app->share(function ($app)
+        {
 
             // Instantiate the AWS service builder
-            $aws = Aws::factory($config);
+            $aws = Aws::factory($app['config']->get('aws'));
 
             // Attach an event listener that will append the Laravel and module version numbers to the user agent string
-            $aws->getEventDispatcher()->addListener('service_builder.create_client', function (Event $event) {
-                $clientConfig = $event['client']->getConfig();
-                $commandParams = $clientConfig->get(Client::COMMAND_PARAMS) ?: array();
-                $userAgentSuffix = 'Laravel/' . Application::VERSION . ' L4MOD/' . AwsServiceProvider::VERSION;
-                $clientConfig->set(Client::COMMAND_PARAMS, array_merge_recursive($commandParams, array(
+            $aws->getEventDispatcher()->addListener('service_builder.create_client', function (Event $event)
+            {
+                $clientConfig       = $event['client']->getConfig();
```

Comment:
Sorry, used Sublime Text for the first time in a while and its seemed to cause all sorts of indentation issues. I'll fix it in a few minutes.